### DISCUSSION TITLE
useAppStoreWithOut 和 useAppStore 的区别？

### DISCUSSION THREAD

**Discussion by aimm** [vbenjs/vue-vben-admin#1809] [Q&A] [answered]
// Need to be used outside the setup
export function useAppStoreWithOut() {
  return useAppStore(store);
}

useAppStoreWithOut 内部也是调用一下 useAppStore，那在外界调用 useAppStoreWithOut 和 直接调用 useAppStore 有什么区别啊

**Comment by zhangyuwaitao** [comment]
我也想知道为什么

**Comment by likui628** [comment, marked as answer]
https://pinia.vuejs.org/core-concepts/outside-component-usage.html
pinia可以在组件内调用，在组件外调用时你得保证pinia被激活。
这里参考了[Using-the-store-outside-of-setup-](https://pinia.vuejs.org/ssr/nuxt.html#Using-the-store-outside-of-setup-)的做法，将pinia的实例传入useStore
```js
import { useStore } from '~/stores/myStore'

export default {
  asyncData({ $pinia }) {
    const store = useStore($pinia)
  },
}
```
